src/hooks/useCollection.js

```diff
@@ -15,48 +15,62 @@ import firestore from 'services/firebase/firestore'
 const baseSortRule = ['_createdAt', 'desc']
 
 const useCollection = (props) => {
+  const {
+    ref: propsRef,
+    where: propsWhere,
+    orderBy: propsOrderBy,
+    limit: propsLimit
+  } = props
   const [value, setValue] = useState()
-  const [loading, setLoading] = useState(true)
+  const [loading, setLoading] = useState(!!propsRef)
   const [loadingMore, setLoadingMore] = useState()
   const [error, setError] = useState()
   const [lastVisible, setLastVisible] = useState()
   const [loadMoreAvailable, setLoadMoreAvailable] = useState(true)
 
   useEffect(() => {
-    if (props?.where?.length) {
+    if (propsWhere?.length) {
       setLoadMoreAvailable(false)
     }
-  }, [props?.where])
+  }, [propsWhere])
 
   useEffect(() => {
     const fetchData = async () => {
       setLoading(true)
-      const documents = []
       try {
-        const querySnapshot = props?.where?.length
+        const querySnapshot = propsWhere?.length
           ? query(
-              collection(firestore, props?.ref),
-              ...props?.where.map((rule) => where(...rule))
+              collection(firestore, propsRef),
+              ...propsWhere.map((rule) => where(...rule)),
+              Array.isArray(propsOrderBy)
+                ? orderBy(...propsOrderBy)
+                : orderBy(...baseSortRule),
+              limit(propsLimit)
             )
           : query(
-              collection(firestore, props?.ref),
-              Array.isArray(props?.orderBy)
-                ? orderBy(...props?.orderBy)
+              collection(firestore, propsRef),
+              Array.isArray(propsOrderBy)
+                ? orderBy(...propsOrderBy)
                 : orderBy(...baseSortRule),
-              limit(props?.limit)
+              limit(propsLimit)
             )
         onSnapshot(querySnapshot, (data) => {
           setLoading(true)
-          setValue(data.docs.map((item) => item?.data()))
+          const documents = data?.docs?.map((item) => item?.data())
+          setValue(documents)
+          if (propsLimit) {
+            setLastVisible(data?.docs?.[data?.docs?.length - 1])
+          }
           setLoading(false)
         })
       } catch (err) {
+        // eslint-disable-next-line no-console
+        console.log(err)
```

Comment:
There is special type of console output function for the errors, - `console.error()`...